pkg/operator/ceph/object/objectstore.go

```diff
@@ -88,18 +98,91 @@ func deleteRealmAndPools(context *Context, spec cephv1.ObjectStoreSpec) error {
 	return nil
 }
 
+func checkZoneIsMaster(context *Context, realmArg string, zoneGroupArg string, zoneArg string) (bool, error) {
+	zoneGroupOutput, err := RunAdminCommandNoRealm(context, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone group")
+	}
+	zoneGroupJson, err := decodeZoneGroupConfig(zoneGroupOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse master zone id")
+	}
+
+	zoneOutput, err := RunAdminCommandNoRealm(context, "zone", "get", realmArg, zoneGroupArg, zoneArg)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get rgw zone")
+	}
+	zoneID, err := decodeID(zoneOutput)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to parse zone id")
+	}
+
+	if zoneID == zoneGroupJson.MasterZoneID {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func getZoneEndpoints(context *Context, realmArg string, zoneGroupArg string, zoneName string, serviceEndpoint string) (string, error) {
+	zoneGroupOutput, err := RunAdminCommandNoRealm(context, "zonegroup", "get", realmArg, zoneGroupArg)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get rgw zone group")
+	}
+	zoneGroupJson, err := decodeZoneGroupConfig(zoneGroupOutput)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse zones list")
+	}
+
+	zoneEndpoints := []string{}
+	for _, zone := range zoneGroupJson.Zones {
+		if zone.Name == zoneName {
+			for _, endpoint := range zone.Endpoints {
+				// in case object-store operator code is rereconciled, zone modify could get run again with serviceEndpoint added again
+				if endpoint != serviceEndpoint {
+					zoneEndpoints = append(zoneEndpoints, endpoint)
+				}
+			}
+			break
+		}
+	}
+
+	zoneEndpoints = append(zoneEndpoints, serviceEndpoint)
+
+	return strings.Join(zoneEndpoints, ","), nil
+}
+
 func setMultisite(context *Context, serviceIP string, spec cephv1.ObjectStoreSpec, realmName string, zoneGroupName string, zoneName string) error {
 	realmArg := fmt.Sprintf("--rgw-realm=%s", realmName)
 	zoneGroupArg := fmt.Sprintf("--rgw-zonegroup=%s", zoneGroupName)
 	zoneArg := fmt.Sprintf("--rgw-zone=%s", zoneName)
-	endpointArg := fmt.Sprintf("--endpoints=%s:%d", serviceIP, spec.Gateway.Port)
 	updatePeriod := false
+	serviceEndpoint := fmt.Sprintf("http://%s:%d", serviceIP, spec.Gateway.Port)
```

Comment:
Soon with https://github.com/rook/rook/pull/5774, we will have a function for this, just wait for a little.

Reply:
Also, using the internal is probably better.